crates/ide-assists/src/handlers/extract_module.rs

```diff
@@ -175,31 +175,36 @@ fn generate_module_def(
     parent_impl: &Option<ast::Impl>,
     module: &mut Module,
     old_indent: IndentLevel,
-) -> String {
-    let (items_to_be_processed, new_item_indent) = if parent_impl.is_some() {
-        (Either::Left(module.body_items.iter()), old_indent + 2)
+) -> ast::Module {
+    let items = if let Some(self_ty) = parent_impl.as_ref().and_then(|imp| imp.self_ty()) {
+        let assoc_items = module
+            .body_items
+            .iter()
+            .filter_map(|item| match item {
+                ast::Item::Const(it) => Some(ast::AssocItem::Const(it.clone())),
+                ast::Item::Fn(it) => Some(ast::AssocItem::Fn(it.clone())),
+                ast::Item::MacroCall(it) => Some(ast::AssocItem::MacroCall(it.clone())),
+                ast::Item::TypeAlias(it) => Some(ast::AssocItem::TypeAlias(it.clone())),
+                _ => None,
+            })
```

Comment:
```suggestion
        let body_items = std::mem::take(&mut module.body_items);
        let assoc_items = body_items
            .into_iter()
            .filter_map(ast::AssocItem::cast)
```

since it doesn't seem like we use `body_items` after this.